imageprocessing/src/main/java/com/example/imageprocessing/PreProcessing.java

```diff
@@ -0,0 +1,394 @@
+        //Converts the image into a matrix
+        Mat brightnessMat = new Mat();
+
+        //Changes the format of the matrix into a RGB one
+        Imgproc.cvtColor(imageMat, brightnessMat, Imgproc.COLOR_RGBA2RGB);
+
+        //Obtain 3 different matrix with the 3 elemental colors
+        List<Mat> color = new ArrayList<>();
+        Core.split(brightnessMat, color);
+
+        /*Each color is multiplied with his luminance.
+          The colors are in order RGB, so to access the che color I use the number 0, 1, 2 in order
+          For more informations see https://en.wikipedia.org/wiki/Relative_luminance*/
+        Mat redLuminance = new Mat();
+        Core.multiply(color.get(0), new Scalar(0.2126), redLuminance);
+        Mat greenLuminance= new Mat();
+        Core.multiply(color.get(1), new Scalar(0.7152), greenLuminance);
+        Mat blueLuminance = new Mat();
+        Core.multiply(color.get(2), new Scalar(0.0722), blueLuminance);
+
+        //Sums the matrix of the colors into a single one
+        Mat tempLuminance = new Mat();
+        Mat totalLuminance = new Mat();
+        Core.add(redLuminance , greenLuminance , tempLuminance); //Red + Green = Temp
+        Core.add(tempLuminance , blueLuminance , totalLuminance); //Temp + Blue = Luminance
+
+        //Calculate the sum of the values of all pixels
+        Scalar sum = Core.sumElems(totalLuminance);
+
+        //Determines the image's bit
+        int bit;
+        switch ( brightnessMat.depth() ) {
+            case CV_8U:  bit = 8; break;
+            case CV_8S:  bit = 8; break;
+            case CV_16U: bit = 16; break;
+            case CV_16S: bit = 16; break;
+            case CV_32S: bit = 32; break;
+            case CV_32F: bit = 32; break;
+            case CV_64F: bit = 64; break;
+            default: return BrightnessValue.IMAGE_IS_OK;
+        }
+        //Calculate the percentage of the brightness
+        double brightness = sum.val[0]/((Math.pow(2,bit)-1)*brightnessMat.rows()
+                *brightnessMat.cols())*2;
+
+        Log.d(TAG, "Brightness:"+brightness);
+
+        /*Bounds to define if the image is dark or bright.
+          The values were decided on the basis of various tests*/
+        double upperBound = 0.9;
+        double lowerBound = 0.4;
+
+        if (brightness > upperBound){             //Image is too bright
+            return BrightnessValue.IMAGE_TOO_BRIGHT;
+        } else if (brightness < lowerBound){        //Image is too dark
+            return BrightnessValue.IMAGE_TOO_DARK;
+        } else {      //Image is neither too bright nor too dark
+            return BrightnessValue.IMAGE_IS_OK;
+        }
+    }
+
+
+    /**
+     * Change the brightness of the image into an optimal one
+     * @param image the image we want to modify the brightness
+     * @return the image with the modified brightness
+     * @author Thomas Porro(g1), Giovanni Fasan(g1), Oscar Garrido(g1)
+     */
+    private Bitmap editBright(Bitmap image){
```

Comment:
I suggest to change the name of the method to something like "autoAdjustImageBrightness" because I would use "editBright" for methods where I have to tell the new brightness.